aws/modules/infrastructure_modules/eks/variables.tf

```diff
@@ -95,3 +95,24 @@ variable "eks_node_tenancy" {
     error_message = "Value must be one of 'default', 'dedicated', or 'host'."
   }
 }
+variable "enable_cis_bootstrap" {
+  description = "Set to true to enable the CIS Boostrap, false to disable."
+  type        = bool
+  default     = false
+}
+
+
+variable "cis_bootstrap_image" {
+  description = "CIS Bootstrap image, required if enable_cis_bootstrap is set to true"
+  type = string
+  default = ""
+}
+
+resource "null_resource" "cis_bootstrap_validation" {
+  lifecycle {
+    precondition {
+      condition     = (var.enable_cis_bootstrap == true && var.cis_bootstrap_image != "") || var.enable_cis_bootstrap == false
+      error_message = "cis_bootstrap_image must be set if enable_cis_bootstrap is true"
+    }
+  }
```

Comment:
That's an interesting way of handling validation of multiple variables. Is there a convention on where this should go in TF?

I like the idea of this, but possibly put it into a new file of extra validations...

(Any ideas @russellseymour?)

Reply:
Not aware of any conventions around this, but agree could make sense to have it in separate validation tf file. Happy to go with whatever you guys think.